lib/improver/metadata/utilities.py

```diff
@@ -64,6 +73,19 @@ def create_new_diagnostic_cube(
         iris.cube.Cube:
             Cube with correct metadata to accommodate new diagnostic field
     """
+    attributes = mandatory_attributes
+    if optional_attributes is not None:
+        for attr in optional_attributes:
```

Comment:
Can't you just do this with `attributes.update(optional_attributes)`?

Reply:
Thanks - done.